src/ucp/core/ucp_context.c

```diff
@@ -28,19 +28,26 @@ static const char *ucp_atomic_modes[] = {
     [UCP_ATOMIC_MODE_LAST]   = NULL,
 };
 
+static const char * ucp_device_type_names[] = {
+    [UCT_DEVICE_TYPE_NET]  = "network",
+    [UCT_DEVICE_TYPE_SHM]  = "intra-node",
+    [UCT_DEVICE_TYPE_ACC]  = "accelerator",
+    [UCT_DEVICE_TYPE_SELF] = "loopback",
+};
```

Comment:
the dimension already exists (in UCT), users can specify devices for each type.
this PR just add more verbose error message if devices are not available.